crates/router/src/core/errors/utils.rs

```diff
@@ -13,6 +13,7 @@ pub trait StorageErrorExt {
 }
 
 impl StorageErrorExt for error_stack::Report<errors::StorageError> {
+    #[track_caller]
     fn to_not_found_response(
```

Comment:
Is the `#[track_caller]` annotation only required for this method, what about other extension trait methods? Or was this added for testing?

Reply:
We actually need to add this where ever we are going change context as this will actually point to the call site

Reply:
Please make the necessary changes in that case. The attribute macro should help us debug errors better.

Reply:
Yes, I am considering to use it in the places where the error were handled declaratively, to make the location more apparent.

Reply:
I'll include these changes in the subsequent PRs